### PR TITLE
add bd classes

### PR DESCRIPTION
**Pull request by LozovskiM**

### CODE REVIEW

File: Cinema system/Cinema system/Models/Cinema.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Cinema_system.Models
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Cinema.cs]: Use CamelCase in namespaces. Also preferably stick to using [Organization].[Product].[Namespace].[SubNameSpace].[SubSubNamespace].[...] pattern

File: Cinema system/Cinema system/Models/Cinema.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    public partial class Cinema
+    {
+        public Cinema()
+        {
+            Hall = new HashSet<Hall>();
+        }
+
+        public int Id { get; set; }
+        public string Title { get; set; }
+        public int NumOfHalls { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Cinema.cs]: Prefer to use full words in names, e.g. NumberOfHalls

File: Cinema system/Cinema system/Models/CinemaDBContext.cs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        public virtual DbSet<Seance> Seance { get; set; }
+        public virtual DbSet<SeanceSeats> SeanceSeats { get; set; }
+        public virtual DbSet<Seat> Seat { get; set; }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+#warning To protect potentially sensitive information in your connection string, you should move it out of source code. See http://go.microsoft.com/fwlink/?LinkId=723263 for guidance on storing connection strings.
+                optionsBuilder.UseSqlServer("Data Source=LAPTOP-HTQV3HQS\\MSSQLSERVER01;Initial Catalog=CinemaDB;Integrated Security=True;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False");
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/CinemaDBContext.cs]: Move connection string to appsettings.json. Use standard configuration tools to access it on application startup.

File: Cinema system/Cinema system/Models/CinemaDBContext.cs [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            }
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Cinema>(entity =>
+            {
+                entity.Property(e => e.Id)
+                    .HasColumnName("ID")
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/CinemaDBContext.cs]: Why have different column name in database?

File: Cinema system/Cinema system/Models/CinemaDBContext.cs [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                    .IsUnicode(false);
+            });
+
+            modelBuilder.Entity<Hall>(entity =>
+            {
+                entity.Property(e => e.Id)
+                    .HasColumnName("ID")
+                    .ValueGeneratedNever();
+
+                entity.Property(e => e.CinemaId).HasColumnName("CinemaID");
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/CinemaDBContext.cs]: Start every method call from new line when using fluent syntax.

File: Cinema system/Cinema system/Models/CinemaDBContext.cs [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+                entity.HasOne(d => d.Cinema)
+                    .WithMany(p => p.Hall)
+                    .HasForeignKey(d => d.CinemaId)
+                    .OnDelete(DeleteBehavior.ClientSetNull)
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/CinemaDBContext.cs]: Why using this exact strategy? Are you going to have halls without cinemas? What is overall record deletion strategy for your application?

File: Cinema system/Cinema system/Models/Cinema.cs (hunk 2 above)
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Cinema.cs]: What is the purpose of this field?

File: Cinema system/Cinema system/Models/Cinema.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        public int NumOfSeats { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Cinema.cs]: What is the purpose of this field?

File: Cinema system/Cinema system/Models/Hall.cs [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public partial class Hall
+    {
+        public Hall()
+        {
+            Seat = new HashSet<Seat>();
+        }
+
+        public int Id { get; set; }
+        public int NumOfSeats { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Hall.cs]: What is the purpose of this field?

File: Cinema system/Cinema system/Models/Movie.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    public partial class Movie
+    {
+        public Movie()
+        {
+            Seance = new HashSet<Seance>();
+        }
+
+        public int Id { get; set; }
+        public string Title { get; set; }
+        public DateTime RealiseDate { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Movie.cs]: Did you mean "ReleaseDate"?

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]
Do you really need all the files in this commit?

File: Cinema system/Cinema system/Models/CinemaDBContext.cs (hunk 4 above)
Comment (andrei-kruhliakou-itechart) [reply Cinema system/Cinema system/Models/CinemaDBContext.cs]: It's better to use "Id" in database instead. Also do you need to call HasColumnName now when names are the same?

File: Cinema system/Cinema system/Models/Cinema.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+        {
+            Hall = new HashSet<Hall>();
+        }
+
+        public int Id { get; set; }
+        public string Title { get; set; }
+        public string City { get; set; }
+        public bool IsDeleted { get; set; }
+
+        public virtual ICollection<Hall> Hall { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Cinema.cs]: Collection name should be in plural form, e.g. "Halls"

File: Cinema system/Cinema system/Models/CinemaDBContext.cs [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+{
+    public partial class CinemaDBContext : DbContext
+    {
+
+        public CinemaDBContext(DbContextOptions<CinemaDBContext> options)
+            : base(options)
+        {
+        }
+
+        public virtual DbSet<Cinema> Cinema { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/CinemaDBContext.cs]: Collection names should be in plural form, e.g. "Cinemas"

File: Cinema system/Cinema system/Models/SeanceSeat.cs [hunk 12]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CinemaSystem.Models
+{
+    public partial class SeanceSeat
+    {
+        public int SeanceId { get; set; }
+        public int SeatId { get; set; }
+        public double Price { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/SeanceSeat.cs]: Do not use types with floating point for money, use decimal instead.

File: Cinema system/Cinema system/Models/SeanceSeat.cs [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+        public bool Available { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/SeanceSeat.cs]: Boolean properties should in general use Is or Has prefix, e.g "IsAvailable", "HasEmptySeats", etc

File: Cinema system/Cinema system/Models/Seat.cs [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+{
+    public partial class Seat
+    {
+        public Seat()
+        {
+            Order = new HashSet<Order>();
+        }
+
+        public int Id { get; set; }
+        public string TypeOfSeat { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Seat.cs]: Why use string for this property? If you have fixed list of possible values it's much better to use Enum or lookup table.

File: Cinema system/Cinema system/Models/Seat.cs [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+        public int Row { get; set; }
+        public int Place { get; set; }
+        public int HallId { get; set; }
+        public bool IsDeleted { get; set; }
+
+        public virtual Hall Hall { get; set; }
+        public virtual ICollection<Order> Order { get; set; }
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Seat.cs]: What is the purpose of this field? You can use one way navigation properties, e.g. Order having collection of Seats makes sense, but Seat having collection of Orders most probably not.

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]
Please add .vs folder in .gitignore, all those files should not be uploaded to git repo.

File: Cinema system/Cinema system/Models/Seat.cs [hunk 16]
```diff
@@ -5,19 +5,21 @@ namespace CinemaSystem.Models
...
     {
-        public Seat()
-        {
-            Order = new HashSet<Order>();
-        }
 
         public int Id { get; set; }
-        public string TypeOfSeat { get; set; }
+
+        public enum SeatType
```
Comment (andrei-kruhliakou-itechart) [inline Cinema system/Cinema system/Models/Seat.cs]: Can you please move it outside of a class?

File: Cinema system/Cinema system/Models/SeanceSeat.cs (hunk 12 above)
Comment (andrei-kruhliakou-itechart) [reply Cinema system/Cinema system/Models/SeanceSeat.cs]: How are you going to handle prices with fractions, like 2.50?

File: Cinema system/Cinema system/Models/CinemaDBContext.cs (hunk 11 above)
Comment (andrei-kruhliakou-itechart) [reply Cinema system/Cinema system/Models/CinemaDBContext.cs]: This is still not fixed

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]
I can still see files from .vs folder in the commit, can you please delete them?

**Review by LozovskiM — COMMENTED** [review]
Sorry, forgot to save gitignore to exclude .vs folders. I'll do it in the next commit.

**Review by andrei-kruhliakou-itechart — APPROVED** [review]

**Pull request merged**